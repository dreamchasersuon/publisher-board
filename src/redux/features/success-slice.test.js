import configureStore from 'redux-mock-store';
import success, { setSuccessMessage } from './successFeatureSlice';

describe('redux set and remove success messages', () => {
  test('action with success message', () => {
    const mockStore = configureStore();
    const store = mockStore({});
    store.dispatch(
      setSuccessMessage({
        message: 'User vasyan_13 updated',
      }),
    );
    expect(store.getActions()).toMatchSnapshot();
  });

  test('reducer should handle setSuccessMessage action', () => {
    expect(
      success([], {
        type: 'success/setSuccessMessage',
        payload: {
          message: 'User vasyan_13 updated',
        },
      }),
    ).toMatchSnapshot();
  });

  test('reducer should handle removeSuccessMessage action', () => {
    expect(
      success(['User vasyan_13 updated'], {
        type: 'success/removeSuccessMessage',
      }),
    ).toMatchSnapshot();
  });
});
