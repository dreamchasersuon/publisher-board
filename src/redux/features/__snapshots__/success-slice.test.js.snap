// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux set and remove success messages action with success message 1`] = `
Array [
  Object {
    "payload": Object {
      "message": "User vasyan_13 updated",
    },
    "type": "success/setSuccessMessage",
  },
]
`;

exports[`redux set and remove success messages reducer should handle removeSuccessMessage action 1`] = `Array []`;

exports[`redux set and remove success messages reducer should handle setSuccessMessage action 1`] = `
Array [
  "User vasyan_13 updated",
]
`;
